addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +120,134 @@ def _get_duration_from_tracking(self, trackings):
+        rot_enabled = self.filtered_domain(self._get_rotting_domain())
+        others = self - rot_enabled
+        for stage, records in rot_enabled.grouped(self._track_duration_field).items():
+            rotting = records.filtered(lambda record: record.date_last_stage_update + timedelta(days=stage.rotting_threshold_days) < now)
+            for record in rotting:
+                record.is_rotting = True
+                record.rotting_days = (now - record.date_last_stage_update).days
+            others += records - rotting
+        others.update({
+            'is_rotting': False,
+            'rotting_days': 0,
+        })
+
+    def _search_is_rotting(self, operator, value):
+        """
+        :param operator
+        :param value
+        :return: domain
+        """
+        if operator not in ['in', 'not in']:
+            raise UserError(_('Operation not supported'))
+        if not self._is_rotting_feature_enabled():
+            raise UserError(_('Model configuration does not support the rotting feature'))
+        base_query = self._search(self._get_rotting_domain())
+
+        # Our query needs to JOIN the stage field's table.
+        # This JOIN needs to use the same alias as the base query to avoid non-matching alias issues
+        # Note that query objects do not make their alias table available trivially,
+        # but the alias can be inferred by consulting the _joins attribute and compare it to the result of make_alias()
+        stage_table_alias_name = base_query.make_alias(self._table, self._track_duration_field)
+
+        # We only need to add a JOIN if the stage table is not already present in the query's _joins attribute.
+        from_add_join = ''
+        if not base_query._joins or not stage_table_alias_name in base_query._joins:
+            from_add_join = """
+                INNER JOIN %(stage_table)s AS %(stage_table_alias_name)s
+                    ON %(stage_table_alias_name)s.id = %(table)s.%(stage_table_alias_name)s
+            """
+
+        # Items with a date_last_stage_update inferior to that number of months will not be returned by the search function.
+        max_rotting_months = int(self.env['ir.config_parameter'].sudo().get_param('crm.lead.rot.max.months', default=12))
+
+        query = f"""
+            WITH perishables AS (
+                SELECT  %(table)s.id AS id,
+                        (
+                            %(table)s.date_last_stage_update + %(stage_table_alias_name)s.rotting_threshold_days * interval '1 day'
+                        ) AS date_rot
+                FROM %(from_clause)s
+                    {from_add_join}
+                WHERE
+                    %(table)s.date_last_stage_update > %(today)s - INTERVAL '%(max_rotting_months)s months'
+                    AND %(where_clause)s
+            )
+            SELECT id
+            FROM perishables
+            WHERE %(today)s >= date_rot
+
+        """
+        self.env.cr.execute(SQL(query,
```

Comment:
should probably flush some records/fields before doing manual query

flush should never be in tests but in actual code